app/root.tsx

```diff
@@ -142,6 +144,13 @@ function Document({
 }
 
 export default function App() {
+  let ranEffect = false;
+  useEffect(() => {
+    if (!ranEffect) {
+      initializeMatomoTagManager("SmoWWpiD");
+      ranEffect = true;
+    }
+  }, []);
```

Comment:
The below code should have the same behavior. 

An empty dependency array makes the `useEffect` [run only once](https://react.dev/reference/react/useEffect#examples-dependencies) (on component mount). This should be why `initializeMatomoTagManager` is only running once. 

The `ranEffect` actually is probably trying to rerun the `initializeMatomoTagManager`. This is because it is not wrapped in 'useState'. Instead it is directly defined within the component. Consequently, every time the component re-renders, `ranEffect` is reinitialized and set to `false`. 

Fortunately, the `useEffect` only runs once, regardless of component rerenders, because the dependency array is empty. Consequently, the `if(!ranEffect)` is never run a second time.

If you do see the `initializeMatomoTagManager` run twice in development, it's a `dev mode` feature meant to test for memory leaks. It will truly only run once in production.

```suggestion
  useEffect(() => {
      initializeMatomoTagManager("SmoWWpiD");
  }, []);
```